### PR TITLE
[JENKINS-48932] - Update plugin to make it runnable with PCT

### PR DESCRIPTION
**Pull request by oleg-nenashev**
I was trying to make the plugin runnable with PCT in order to have a test suite for https://issues.jenkins-ci.org/browse/JENKINS-48932 . The codebase is pretty obsolete, and the tests hang on my machine IIUC. Let's see whether tests pass in CI.

@reviewbybees @rsandell

### CODE REVIEW

**Comment by rsandell** [comment]
I have started a similar effort on the [newpom](https://github.com/jenkinsci/build-failure-analyzer-plugin/tree/newpom) branch. But so far your effort seems a bit better than my attempt :)

**Comment by oleg-nenashev** [comment]
Some bits still need to be cleared up: `java.lang.NoClassDefFoundError: Could not initialize class org.mockito.internal.creation.jmock.ClassImposterizer`

**Comment by oleg-nenashev** [comment]
Alternate fix on the core side: https://github.com/jenkinsci/jenkins/pull/3234

**Comment by oleg-nenashev** [comment]
Apparently the tests are green.

File: src/test/java/com/sonyericsson/jenkins/plugins/bfa/model/indication/MultilineBuildLogIndicationTest.java [hunk 1]
```diff
@@ -176,13 +195,14 @@ public void testDoMatchTextUrlValidOkMatrixProject() throws Exception {
...
+    @Test
     public void testDoMatchTextUrlValidWarning() throws Exception {
-        FreeStyleProject freeStyleProject = createFreeStyleProject();
+        FreeStyleProject freeStyleProject = j.createFreeStyleProject();
         freeStyleProject.getBuildersList().add(new PrintToLogBuilder(TEST_STRING));
         FreeStyleBuild freeStyleBuild = freeStyleProject.scheduleBuild2(0, new Cause.UserCause()).get();
         MultilineBuildLogIndication.MultilineBuildLogIndicationDescriptor indicationDescriptor =
                 new MultilineBuildLogIndication.MultilineBuildLogIndicationDescriptor();
-        String buildUrl = getURL() + freeStyleBuild.getUrl();
+        String buildUrl = j.getURL() + freeStyleBuild.getUrl();
```
Comment (rsandell) [inline src/test/java/com/sonyericsson/jenkins/plugins/bfa/model/indication/MultilineBuildLogIndicationTest.java]: Line 215: missing `@Test`

File: pom.xml [hunk 2]
```diff
@@ -216,6 +233,12 @@
             <version>${powermock.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-core</artifactId>
```
Comment (jglick) [inline pom.xml]: Usually done by excluding the wrong dependency chain.

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/dbf/DownstreamBuildFinder.java [hunk 3]
```diff
@@ -38,7 +38,6 @@
 /**
  * There is no general way to find downstream build in Jenkins Core. Different
  * plugin have there own way of keeping this information.
- * <p/>
```
Comment (jglick) [inline src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/dbf/DownstreamBuildFinder.java]: just <p> is enough

**Comment by oleg-nenashev** [comment]
Tests seem to be flaky

screen shot 2018-01-15 at 18 42 03

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/dbf/DownstreamBuildFinder.java (hunk 3 above)
Comment (jglick) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/dbf/DownstreamBuildFinder.java]: `<p>` that is

**Comment by rsandell** [comment]
Well actually not merge until snapshot dep is fixed

File: pom.xml [hunk 4]
```diff
@@ -77,21 +77,26 @@
...
+            <url>https://repo.jenkins-ci.org/public</url>
         </pluginRepository>
     </pluginRepositories>
 
     <dependencies>
         <dependency>
             <groupId>com.sonyericsson.hudson.plugins.gerrit</groupId>
             <artifactId>gerrit-trigger</artifactId>
-            <version>2.15.0</version>
+            <version>2.27.2-20180115.173211-1</version>
```
Comment (rsandell) [inline pom.xml]: 2.27.2 is now released

**Pull request merged**